File: CheckOCR/CheckOCR/CheckResult.cpp
//
//  CheckResult.cpp
//  CheckOCR
//
//  Created by Viktor Levshchanov on 22.11.14.
//  Copyright (c) 2014 Viktor Levshchanov. All rights reserved.
//

#include "CheckResult.h"
#include "CheckResultComponent.h"

namespace CheckOCR {
    
    CheckResult::CheckResult(CheckResultStatus status, int errorCode) : _status(status), _errorCode(errorCode)
    {
    };
    
    CheckResult::CheckResult(ComponentType componentsType, const std::vector<CheckResultComponent> components) :
    _status(CHECK_OCR_SUCCESS), _componentsType(componentsType), _components(components)
    {
    };
    
    CheckResult::~CheckResult() { }
    
    CheckResultStatus CheckResult::GetStatus()
    {
        return _status;
    }
    
    int CheckResult::GetErrorCode()
    {
        return _errorCode;
    }
    
    ComponentType CheckResult::GetComponentsType()
    {
        return _componentsType;
    }
    
    const std::vector<CheckResultComponent> CheckResult::GetComponents()
    {
        return _components;
    }
    
    bool CheckResult::IsSuccess()
    {
        return _status == CHECK_OCR_SUCCESS && _errorCode == 0;
    }
    
}

File: CheckRecognition/CheckRecognition/Model/CheckResult/CheckResult.swift
//
//  CheckResult.swift
//  CheckRecognition
//
//  Created by Viktor Levshchanov on 3/6/15.
//  Copyright (c) 2015 Viktor Levshchanov. All rights reserved.
//

import UIKit

class CheckResult: NSObject {
   
    let components: [CheckResultComponent]
    
    init(components: [AnyObject]) {
        self.components = components as [CheckResultComponent] //TODO: check if this line is safe
        super.init()
    }
}

class MutableCheckResult: NSObject {
    var components: [CheckResultComponent]
    
    override init() {
        components = [CheckResultComponent]()
        super.init()
    }
    
    func addComponent(component: CheckResultComponent) {
        components.append(component)
    }
    
    func result() -> CheckResult {
        return CheckResult(components: components)
    }
}